File: Game/src/main/java/net/winternetwork/bedwars/game/modules/map/MapModule.java
package net.winternetwork.bedwars.game.modules.map;

import lombok.Getter;
import net.winternetwork.bedwars.api.module.Module;
import net.winternetwork.bedwars.api.module.ModulePriority;
import net.winternetwork.bedwars.game.modules.map.command.MapSetupCommand;
import net.winternetwork.bedwars.game.net.winternetwork.bedwars.game.Game;

public class MapModule extends Module {

    @Getter
    private MapManager mapManager;

    public MapModule() {
        super("Mapa", ModulePriority.NORMAL);
    }

    @Override
    public void init() {
        mapManager = new MapManager();

        log("Carregando mapas...");
        mapManager.loadAll(Game.getGame().getMapsConfig());

        log("Registrando comandos...");
        Game.getGame()
                .registerCommands(new MapSetupCommand(mapManager));
    }

    @Override
    public void disable() {
        log("Salvando os mapas...");
        mapManager.unloadAll(Game.getGame().getMapsConfig());
    }
}

File: Game/src/main/kotlin/net/winternetwork/bedwars/game/modules/maps/MapModule.kt
package net.winternetwork.bedwars.game.modules.maps

import net.winternetwork.bedwars.api.config.YamlConfig
import net.winternetwork.bedwars.api.module.Module
import net.winternetwork.bedwars.game.game
import net.winternetwork.bedwars.game.modules.maps.command.MapCommands
import net.winternetwork.bedwars.game.modules.maps.manager.MapManager

class MapModule : Module("Maps") {

    val mapManager = MapManager()

    private val config: YamlConfig by lazy {
        YamlConfig(
                game,
                dataFolder,
                "maps.yml"
        )
    }

    override fun init() {
        log("Carregando mapas...")
        mapManager.loadAll(config)

        log("Registrando comandos...")
        game.commands(MapCommands(mapManager))
    }

    override fun disable() {
        log("Salvando mapas...")
        mapManager.unloadAll(config)
    }
}